src/wallet/wallet.cpp

```diff
@@ -3975,8 +3975,7 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     // change output. Only try this once.
                     if (nChangePosInOut == -1 && nSubtractFeeFromAmount == 0 && pick_new_inputs) {
                         // Adding another output can change the serialized size of the vout size
-                        unsigned int nBytesCompactSizeDiff = GetSizeOfCompactSize(txNew.vout.size() + 1) - GetSizeOfCompactSize(0);
-                        unsigned int tx_size_with_change = nBytes + change_prototype_size + nBytesCompactSizeDiff;
+                        unsigned int tx_size_with_change = nBytes + change_prototype_size + GetSizeOfCompactSizeDiff(0, txNew.vout.size() + 1);
```

Comment:
I think this should actually be
```suggestion
                        unsigned int tx_size_with_change = nBytes + change_prototype_size + GetSizeOfCompactSizeDiff(txNew.vout.size(), txNew.vout.size() + 1);
```
my bad 🙈